src/rest.rs

```diff
@@ -0,0 +1,115 @@
+use crate::OKEx;
+use serde::{Deserialize, Deserializer, Serialize};
+use std::str::FromStr;
+use std::fmt::Display;
+use std::collections::HashMap;
+
+const API_HOST: &'static str = "https://www.okex.com/api/spot/v3/";
+
+pub struct RestClient {
+    client: reqwest::Client,
+    api_key: String,
+    secret: String,
+}
+
+impl OKEx for RestClient {
+    fn new() -> Self {
+        let timeout = std::time::Duration::from_secs(5);
```

Comment:
Maybe `timeout` should be a constant somewhere?

Reply:
So it would have to be static and even then it may not work cause calls in statics are very limited. You may want to take a look at the lazy_static crate it's quite useful for stuff like this...

Reply:
I meant more the `5` inside of it, but why wouldn't the whole thing work as a static value?

Reply:
It's worth a try in my opinion

Reply:
`calls in statics are limited to constant functions, tuple structs and tuple variants`

It's because destructors cannot be evaluated at compile time. The lazy_static crate basically creates it at run time the first time it is requested then reuses it for all calls (pretty much singleton pattern).

Reply:
Ah, ok.